server/belga/io/feed_parsers/belga_newsml_1_2.py

```diff
@@ -0,0 +1,677 @@
+            if element.get('FormalName', '') is not None:
+                item['comment']['name'] = element.get('FormalName', '')
+
+        # Parser Identification element
+        self.parser_identification(item, newsitem_el.find('Identification'))
+
+        # Parser NewsManagement element
+        self.parser_newsmanagement(item, newsitem_el.find('NewsManagement'))
+
+        # Parser NewsComponent element
+        component_parent = newsitem_el.find('NewsComponent')
+        # if have a child NewsComponent, run the parser for TASS xml file.
+        if component_parent is not None:
+            if component_parent.attrib.get('Duid', '') is not None:
+                item['news_component_duid'] = component_parent.attrib.get('Duid', '')
+
+            if component_parent.attrib.get('Essential', '') is not None:
+                item['news_component_essential'] = component_parent.attrib.get('Essential', '')
+
+            if component_parent.attrib.get('EquivalentsList', '') is not None:
+                item['news_component_equivalentslist'] = component_parent.attrib.get('EquivalentsList', '')
+
+            component_child = component_parent.find('NewsComponent')
+            if component_child is not None:
+                self.parser_newscomponent(item, component_child.find('NewsComponent'))
+            else:
+                self.parser_newscomponent(item, component_parent)
+
+    def parser_identification(self, item, indent_el):
+        """
+        Function parse Identification in NewsItem element.
+
+        Example:
+
+        <Identification>
+          <NewsIdentifier>
+            <ProviderId>afp.com</ProviderId>
+            <DateId>20181209T112417Z</DateId>
+            <NewsItemId>TX-PAR-QCJ26</NewsItemId>
+            <RevisionId PreviousRevision="0" Update="N">1</RevisionId>
+            <PublicIdentifier>urn:newsml:afp.com:20181209T112417Z:TX-PAR-QCJ26:1</PublicIdentifier>
+          </NewsIdentifier>
+          <NameLabel>musique-rock-célébrités-religion-France</NameLabel>
+        </Identification>
+
+        :param item:
+        :param indent_el:
+        :return:
+        """
+        if indent_el is None:
+            return
+
+        newsident_el = indent_el.find('NewsIdentifier')
+        if newsident_el is not None:
+            element = newsident_el.find('ProviderId')
+            if element is not None:
+                item['provide_id'] = element.text
+
+            element = newsident_el.find('DateId')
+            if element is not None:
+                item['date_id'] = element.text
+
+            element = newsident_el.find('NewsItemId')
+            if element is not None:
+                item['item_id'] = element.text
+
+            element = newsident_el.find('RevisionId')
+            if element is not None:
+                item['version'] = element.text
+
+            element = newsident_el.find('PublicIdentifier')
+            if element is not None:
+                item['guid'] = element.text
```

Comment:
I think that we should not use `NewsML->NewsItem->Identification->NewsIdentifier->PublicIdentifier` as `guid`, because it must be unique per item. We should use `NewsComponent`'s `Duid` as unique news identifier (guid).

Reply:
I see EFE,TASS xml have `NewsComponent`'s `Duid`. but AFP, ANP haven't it.
So, I will check if xml file has `NewsComponent`'s `Duid`. we will get it for `guid`. If not we get `PublicIdentifier`. Is it right?
I've checked many xml file. I see `PublicIdentifier` is unique per item.

Reply:
Hi @idsvn , right, please proceed they way you suggested. Do you have Belga's news ml sample files?

Reply:
Here it is:
server/tests/io/fixtures/tass_belga.xml
server/tests/io/fixtures/afp_belga.xml
server/tests/io/fixtures/anp_belga.xml
server/tests/io/fixtures/efe_belga.xml
These are 4 type format of newsml from Belga.
I see a https://dev.sourcefabric.org/browse/SDBELGA-30 .
a other type of newsml Belga, But I don't do this type.